src/typeEvaluator/functions.ts

```diff
@@ -73,6 +73,29 @@ export function handleFuncCallNode(node: FuncCallNode, scope: Scope): TypeNode {
       })
     }
 
+    case 'math.sum':
+    case 'math.avg':
+    case 'math.max':
+    case 'math.min': {
+      const values = walk({node: node.args[0], scope})
+      // use mapConcrete to get concrete resolved value, it will also handle cases where the value is a union
+      return mapConcrete(values, scope, (node) => {
+        // Aggregate functions can only be applied to arrays
+        if (node.type === 'array') {
+          // Resolve the concrete type of the array elements
+          return mapConcrete(node.of, scope, (node) => {
+            // Math functions can only be applied to numbers
+            if (node.type === 'number') {
+              return node
+            }
```

Comment:
This won't be correct when `value` is set and the function is `sum` or `avg`, no?

Reply:
good catch, yess